Fixes string scanning and adds escaping support

artman opened the pull request:
Fixes an issue where double quoted string literals that contained single quotes (or vice verca) failed to parse correctly (e.g. "Isn't").

Adds support for escaping quotes within a string literal (e.g. "This is a \\"test\\"") now parsers correctly.

phated commented:
@YanagiEiichi I think we are fine using the Struct parser and handling "forced optional" in our compiler.  I'll be merging #47 today which will be reflected in my PR.

phated commented:
Handled in the original suggested way and update upstream.

Commit:
Fixes string scanning and adds escaping support

Fixes an issue where double quoted string literals that contained single quotes (or vice verca) failed to parse correctly (e.g. "Isn't").

Adds support for escaping quotes within a string literal (e.g. "This is a \"test\"") now parsers correctly.
